HDOS-480

pka1024 opened the pull request:
feat: (api):
- Workflow operators are working with transformations.
- Fix for category re-rendering after workflow changes.
- Deleted unused component-editor-service.
- Fixed associated unit tests.

Commit:
API change, workflow editor

feat: (api):
- Move and delete workflow operators, in progress.

Commit:
API change, workflow editor

feat: (api):
- Added method getConnectorById.
- Move and delete workflow operators, in progress.

Commit:
Merge branch 'HDOS-307' into HDOS-480

Commit:
API change, workflow editor

feat: (api):
- Fix for category re-rendering after workflow changes.
- Workflow operators, in progress.

Commit:
API change, workflow editor

feat: (api):
- Workflow operators are working with transformations.

Commit:
API change, component-editor-service

feat: (api):
- Deleted unused component-editor-service.
- Fixed associated unit tests.

Commit:
API change, workflow editor

feat: (api):
- Fixed changeRevision dropdown menu filter.

Commit:
API change, workflow editor

feat: (api):
- Added a extended condition for adding links to a workflow.

thatsmeta commented on frontend/src/app/components/workflow-editor/workflow-editor.component.ts:
```diff
@@ -87,8 +90,6 @@ export class WorkflowEditorComponent {
       position
     );
-    // TODO
-    // @ts-ignore
```
Das TODO kann nur entfernt werden, wenn auch unten der Cast entfernt wird.

thatsmeta commented on frontend/src/app/service/type-converter/flowchart-converter.service.ts:
```diff
@@ -241,17 +243,17 @@ export class FlowchartConverterService {
-        } as Point)
+          x: coords[0],
+          y: coords[1]
+        } as Position)
```
Cast sollte nicht mehr notwendig sein.

thatsmeta commented on frontend/src/app/service/type-converter/flowchart-converter.service.ts:
```diff
@@ -224,10 +226,10 @@ export class FlowchartConverterService {
    * @param link svg link element
    */
-  public convertLinkPathToPoints(link: Element): Point[] {
+  public convertLinkPathToPosition(link: Element): Position[] {
```
convertLinkPathToPositions

thatsmeta commented on frontend/src/app/service/type-converter/flowchart-converter.service.ts:
```diff
@@ -277,6 +279,49 @@ export class FlowchartConverterService {
+      );
+    }
+
+    if (foundConnector === undefined) {
```
Block löschen

thatsmeta commented on frontend/src/app/components/workflow-editor/workflow-editor.component.ts:
```diff
@@ -511,7 +548,7 @@ export class WorkflowEditorComponent {
     }
-    this.workflowService.updateWorkflow(this.currentWorkflow);
+    this.baseItemService.updateTransformation(this.currentWorkflow).subscribe();
     this.hasChanges = false;
```
.subscribe(() => this.hasChanges=false)

thatsmeta commented on frontend/src/app/components/workflow-editor/workflow-editor.component.ts:
```diff
@@ -261,31 +259,65 @@ export class WorkflowEditorComponent {
+    let startConnector: Connector;
+    let endConnector: Connector;
+
+    if (linkSourceIds.operatorId === this.currentWorkflow.id) {
```
const sourceIsWorkflowInput = linkSourceIds.operatorId === this.currentWorkflow.id
const startConnector: Connector = this.flowchartConverter.getConnectorFromOperatorById(
        linkSourceIds,
        this.currentWorkflow,
        sourceIsWorkflowInput
      );
const targetIsWorkflowOutput = linkTargetIds.operatorId === this.currentWorkflow.id;
const endConnector: Connector = this.flowchartConverter.getConnectorFromOperatorById(
        linkTargetIds,
        this.currentWorkflow,
        targetIsWorkflowOutput
      );

thatsmeta commented on frontend/src/app/components/workflow-editor/workflow-editor.component.ts:
```diff
@@ -261,31 +259,65 @@ export class WorkflowEditorComponent {
       true
     );
-    const linkTargetIds = this.flowchartConverter.getLinkOperatorAndConnector(
+    const linkTargetIds: VertexIds = this.flowchartConverter.getLinkOperatorAndConnectorId(
```
Link ziehen von WorkflowOutput zu Operator-Output, dann sichergehen, dass der WorkflowOutput hier in den TargetIds steht.

thatsmeta commented on frontend/src/app/service/type-converter/flowchart-converter.service.ts:
```diff
@@ -277,6 +279,49 @@ export class FlowchartConverterService {
   }
 
+  /**
+   * extracts the connector from the given connector id
```
Extract the connector from the operator or workflow given the vertex ids.

thatsmeta commented on frontend/src/app/service/type-converter/flowchart-converter.service.ts:
```diff
@@ -277,6 +279,49 @@ export class FlowchartConverterService {
+  public getConnectorFromOperatorById(
+    vertexIds: VertexIds,
+    workflowTransformation: WorkflowTransformation,
+    isCurrentWorkflowId: boolean
```
searchInWorkflowIoInterface
javadoc: if true, search for the connector in the workflow io interface instead of the workflow content operators

Commit:
API change, workflow editor

feat: (api):
- Added a TODO comment for links.
- Improved the method _checkAndAddLink
